From 133fa187620e31f229560f2ee9115fd542b59994 Mon Sep 17 00:00:00 2001
From: Bin Jiang <bin.jiang@windriver.com>
Date: Thu, 10 Jul 2014 11:22:03 +0800
Subject: [PATCH 415/466] dts/t4240qds: Adjust the partitions of nor flash

Since the size of u-boot provided in sdk1.6 is larger than 0x80000,
adjust partitions of nor flash.

Signed-off-by: Bin Jiang <bin.jiang@windriver.com>
---
 arch/powerpc/boot/dts/t4240qds.dts |   48 +++++++++++++++++++++---------------
 1 file changed, 28 insertions(+), 20 deletions(-)

diff --git a/arch/powerpc/boot/dts/t4240qds.dts b/arch/powerpc/boot/dts/t4240qds.dts
index 9b8a78f..77c333a 100644
--- a/arch/powerpc/boot/dts/t4240qds.dts
+++ b/arch/powerpc/boot/dts/t4240qds.dts
@@ -100,30 +100,30 @@
 				read-only;
 			};
 			partition@20000 {
-				label = "kernel";
-				reg = <0x20000 0x500000>;
+				label = "kernel1";
+				reg = <0x20000 0x7e0000>;
 			};
-			partition@520000 {
-				label = "dtb";
-				reg = <0x520000 0x20000>;
+			partition@800000 {
+				label = "dtb1";
+				reg = <0x800000 0x100000>;
 			};
-			partition@540000 {
+			partition@900000 {
 				label = "fs1";
-				reg = <0x540000 0x3a00000>;
+				reg = <0x900000 0x3600000>;
 			};
-			partition@3f40000 {
+			partition@3f00000 {
 				label = "alt-fman-firmware";
-				reg = <0x3f40000 0x20000>;
+				reg = <0x3f00000 0x20000>;
 				read-only;
 			};
-			partition@3f60000 {
+			partition@3f20000 {
 				label = "alt-uboot-env";
-				reg = <0x3f60000 0x20000>;
+				reg = <0x3f20000 0x20000>;
 				read-only;
 			};
-			partition@3f80000 {
+			partition@3f40000 {
 				label = "alt-u-boot";
-				reg = <0x3f80000 0x80000>;
+				reg = <0x3f40000 0xc0000>;
 				read-only;
 			};
 			partition@4000000 {
@@ -132,22 +132,30 @@
 				read-only;
 			};
 			partition@4020000 {
+				label = "kernel2";
+				reg = <0x4020000 0x7e0000>;
+			};
+			partition@4800000 {
+				label = "dtb2";
+				reg = <0x4800000 0x100000>;
+			};
+			partition@4900000 {
 				label = "fs2";
-				reg = <0x4020000  0x3f20000>;
+				reg = <0x4900000 0x3600000>;
 			};
-			partition@7f40000 {
+			partition@7f00000 {
 				label = "fman-firmware";
-				reg = <0x7f40000 0x20000>;
+				reg = <0x7f00000 0x20000>;
 				read-only;
 			};
-			partition@7f60000 {
+			partition@7f20000 {
 				label = "u-boot env";
-				reg = <0x7f60000  0x20000>;
+				reg = <0x7f20000  0x20000>;
 				read-only;
 			};
-			partition@7f80000 {
+			partition@7f40000 {
 				label = "u-boot";
-				reg = <0x7f80000 0x80000>;
+				reg = <0x7f40000 0xc0000>;
 				read-only;
 			};
 		};
-- 
1.7.10.4

